File: Java/app/src/main/java/com/example/divya/kotlinlearningjava/ui/main/MainFragment.java
package com.example.divya.kotlinlearningjava.ui.main;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;


import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.StaggeredGridLayoutManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.divya.kotlinlearningjava.R;
import com.example.divya.kotlinlearningjava.adapter.CategoryAdapter;
import com.example.divya.kotlinlearningjava.model.Category;

import java.util.ArrayList;
import java.util.List;

public class MainFragment extends Fragment {

    private MainViewModel mViewModel;
    private RecyclerView recyclerView;
    private CategoryAdapter adapter;
    private List<Category> categories = new ArrayList<>();

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.main_fragment, container, false);
        recyclerView = view.findViewById(R.id.categoryList);
        recyclerView.setLayoutManager(new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL));
        adapter = new CategoryAdapter(categories);
        recyclerView.setAdapter(adapter);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = ViewModelProviders.of(this).get(MainViewModel.class);
        mViewModel.init();
        LiveData<List<Category>> liveData = mViewModel.getCategories();
        liveData.observe(getActivity(), (List<Category> categoriesData) -> {
                categories.addAll(categoriesData);
                adapter.notifyDataSetChanged();

        });

    }

}

File: Kotlin/app/src/main/java/com/example/divya/kotlinonthego/ui/main/MainFragment.kt
package com.example.divya.kotlinonthego.ui.main

import android.arch.lifecycle.ViewModelProviders
import android.bluetooth.le.AdvertiseCallback
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.StaggeredGridLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import com.example.divya.kotlinonthego.R
import com.example.divya.kotlinonthego.adapters.CategoryAdapter

class MainFragment : Fragment() {

    companion object {
        fun newInstance() = MainFragment()
    }

    private lateinit var viewModel: MainViewModel
    private lateinit var manager : StaggeredGridLayoutManager

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.main_fragment, container, false)
        viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
        val adapter = CategoryAdapter(viewModel.GetCategorylist(), view.context)
        manager = StaggeredGridLayoutManager(3, StaggeredGridLayoutManager.VERTICAL)
        manager.gapStrategy = StaggeredGridLayoutManager.GAP_HANDLING_NONE
        view.findViewById<RecyclerView>(R.id.category_list).layoutManager = manager
        view.findViewById<RecyclerView>(R.id.category_list).adapter = adapter


        return view
    }


}
